### ISSUE TITLE
Using number as enum variable initializer should be treated as a number rather than literal.

### ISSUE THREAD

**Issue by vivekgalatage** [eslint/typescript-eslint-parser#470]
**What version of TypeScript are you using?**
2.8.1

**What version of `typescript-eslint-parser` are you using?**
15.0.0

**What code were you trying to parse?**
```ts
enum Test { 
  one = 1, 
  two = 1.1, 
  three = 1.2 
}
```

**What did you expect to happen?**
The enum initializer type is number whereas the generated JSON states it to be a string literal.  The typescript compiler treats this as a number only.

**What happened?**
![image](https://user-images.githubusercontent.com/1282684/39178894-83dedea8-47d0-11e8-83dd-d6f5ec4215f0.png)

**Comment by j-f1** [comment]
That’s the correct AST — see [this example](https://astexplorer.net/#/gist/2dc035a24c5ba42cd9e0d21be23622d6/5397499d9e3cc0bfd2c876db401a4b07f62b55fa) with just a number.

**Comment by vivekgalatage** [comment]
This also applies to the decorators where the argument 3.14 is treated as Literal

```ts
function $scope(scope: any[], chart: string, pi: number) {
  return (target: Function) => {}
}

@$scope(['hello'], 'bar', 3.14)
export class ExportedClass {
}
```

**Comment by vivekgalatage** [comment]
This gist [1] specifies the type as `NumericLiteral`

[1] https://astexplorer.net/#/gist/6e9911fc9df41ff778b9c4b90eeafa42/a60adb3256a228b42c31ec9add791676f2512dfe

**Comment by j-f1** [comment]
@vivekgalatage The TypeScript parser doesn’t follow the ESTree spec, which specified that all literales are of the `Literal` type.

**Comment by vivekgalatage** [comment]
@j-f1 sure, thanks for the clarification.  So in that case, how does the developer would differentiate between the numerical values vs the literal values.  Or are we not supposed to use the non literal values at these places?

**Comment by j-f1** [comment]
String literals have a string `value`, and numeric literals have a numeric `value`.

**Comment by vivekgalatage** [comment]
Fantastic!!! And my apologies for missing this info. Thank you!
